btcutil/go.mod

```diff
@@ -4,7 +4,7 @@ go 1.16
 
 require (
 	github.com/aead/siphash v1.0.1
-	github.com/btcsuite/btcd v0.22.0-beta.0.20220111032746-97732e52810c
```

Comment:
This should be able to just jump to v0.23.0 now.

Reply:
Can you run a `go mod tidy` locally? If I do so, I get a small diff that isn't present in this PR. 

Also in the future, the best way to get turn around on a review is comment directly like you did here. Thanks!

Reply:
Thanks. I ran `go mod tidy` with no diffs in the root directory but with some diffs when run from `btcutil/` and `btcetc/` so I added those changes.

(with go version = `go1.18.3`)